### PR TITLE
Add subclasses of BadArgument for converters

### PR DESCRIPTION
**Pull request by muddyfish**
### Summary

This PR adds custom exception classes for converters. This enables the possibility for translation  of these messages, and easier debugging.

Exceptions added for each converter:
 - MemberConverter:
   - MemberNotFound
 - UserConverter:
   - UserNotFound
 - MessageConverter:
   - MessageNotFound
   - ChannelNotFound
   - ChannelNotReadable
 - TextChannelConverter:
   - ChannelNotFound
 - VoiceChannelConverter:
   - ChannelNotFound
 - CategoryChannelConverter:
   - ChannelNotFound
 - ColourConverter:
   - ColourInvalid
 - RoleConverter:
   - RoleNotFound
 - InviteConverter:
   - InviteInvalid
 - EmojiConverter:
   - EmojiNotFound
 - PartialEmojiConverter:
   - PartialEmojiNotFound

### Checklist

- [ ] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/ext/commands/errors.py [hunk 1]
```diff
@@ -202,6 +213,186 @@ class NotOwner(CheckFailure):
...
+
+    .. versionadded:: 1.5
+
+    Attributes
+    -----------
+    member: :class:`str`
+        The member supplied by the caller that was not found
+    """
+    def __init__(self, member):
+        self.member = member
```
Comment (bijij) [inline discord/ext/commands/errors.py:230]: For the sake of custom converters and error handlers which may raise / handle these exceptions.
Could it be wise to rather than uniquely name these attributes based on their relevant argument type, use a more generic Error.argument attribute.

conscider this example:
```py
@bot.event
async def on_command_error(ctx: Context, error: Exception):
    if isinstance(error, (MemberNotFound, UserNotFound)):
        if isinstance(error, MemberNotFound):
            argument = error.member
        elif isinstance(error, UserNotFound):
            argument = error.user
        bot.log.info(f'Converter for command {ctx.command.name} failed, argument supplied: "{argument}"')
```

having all errors share a common attribute name would remove the need for handling individual cases like shown.

Comment (muddyfish) [reply discord/ext/commands/errors.py]: Done, renamed the argument in all converter error classes to `arg`

File: discord/ext/commands/errors.py [hunk 2]
```diff
@@ -202,6 +213,186 @@ class NotOwner(CheckFailure):
...
+    Attributes
+    -----------
+    emoji: :class:`str`
+        The emoji supplied by the caller that did not match the regex
+    """
+    def __init__(self, emoji):
+        self.emoji = emoji
+        super().__init__('Couldn\'t convert "{}" to PartialEmoji.'.format(emoji))
+
+class InvalidBoolean(BadArgument):
```
[the remark this answers is not in the document]
Comment (muddyfish) [reply discord/ext/commands/errors.py:376]: That should now be fixed

**Comment by muddyfish** [comment]
I believe I've made all the changes suggested so far (sorry for the long time)

File: discord/ext/commands/errors.py [hunk 3]
```diff
@@ -43,6 +43,17 @@
...
+    'MemberNotFound',
+    'UserNotFound',
+    'ChannelNotFound',
+    'ChannelNotReadable',
+    'BadColourArgument',
+    'RoleNotFound',
+    'BadInviteArgument',
+    'EmojiNotFound',
+    'PartialEmojiConversionFailure',
+    'InvalidBoolean',
```
Comment (Rapptz) [inline discord/ext/commands/errors.py:56]:
```suggestion
    'BadBoolArgument',
```

Might as well be consistent with it.

**Review by Rapptz — APPROVED** [review]

**Comment by Rapptz** [comment]
Thanks. Though `Boolean` as a name isn't really used within Python so I'll shorten this to `Bool` (i.e. `BadBoolArgument`) in a future commit.
